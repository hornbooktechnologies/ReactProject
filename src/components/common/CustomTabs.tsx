import React from 'react'
import { useTranslation } from 'react-i18next'
import { CustomTabModel } from '../../models/CustomTabModel'

const CustomTabs = (tabObj: CustomTabModel) => {
  const { i18n } = useTranslation();
  return (
    <>
     <ul uk-tab="true" className={['uk-tab', tabObj.customCssClass].join(' ')}>
        { tabObj.tabs?.map((tab: { tabId: string, i8nCode: string }, index) => (
            <li
              key={index}
              className={`${tabObj.selectedTab === tab.tabId ? "uk-active" : ""}`}
              onClick={(e) => {
                  tabObj.onTabSelect.call(this, tab.tabId);
              }}
            >
                <a href="about:blank"> {i18n.t(tab.i8nCode)}</a>
            </li>
        ))}
      </ul>
    </>
  )
}

export default CustomTabs
