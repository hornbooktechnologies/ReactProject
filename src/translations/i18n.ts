import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
