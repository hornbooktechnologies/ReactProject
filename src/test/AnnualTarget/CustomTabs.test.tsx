import React from "react";
import renderer from "react-test-renderer";
import CustomTabs from "../../components/common/CustomTabs";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";

describe("Component: CustomTabs", () => {
  it("CustomTabs renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <CustomTabs />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
