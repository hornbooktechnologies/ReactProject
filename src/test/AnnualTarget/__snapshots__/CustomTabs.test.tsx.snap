// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CustomTabs CustomTabs renders correctly 1`] = `
<ul
  className="uk-tab "
  uk-tab="true"
/>
`;
